### ISSUE TITLE
Outh 2.0 to Sax bank

### ISSUE THREAD

**Issue by MislavSag** [r-lib/httr2#149]
I am trying to authenticate using Outh 2 on Sax Bank. Here are the docs on sax developer site: https://www.developer.saxo/openapi/learn/oauth-authorization-code-grant

I have successfully authorized using python code: 
https://github.com/SaxoBank/openapi-samples-python/tree/master/authentication/oauth
but I can't translate the code in R. I have discovered that redirect url can be set to: http://localhost:4321/redirect

Here is my try:
```
library(httr2)
#> Warning: package 'httr2' was built under R version 4.1.3
authorization_url = "https://sim.logonvalidation.net/authorize"
token_url = "https://sim.logonvalidation.net/token"
client <- oauth_client(
  id = "xxx",
  secret = "xxx",
  token_url = token_url,,
  # key = "3ebb2f9d91664b92b759ddebafe3437b"
  name = "PRA"
)
url_ <- oauth_flow_auth_code_url(client, 
                                 auth_url = authorization_url, 
                                 redirect_uri = "http://localhost:4321/redirect")
# HOW TO EXECUTE ABOVE URL AND HOW TO DEFINE STATE VARIABLE?
```

**Comment by MislavSag** [comment]
After experimenting for some time I have found the solution.

I had to change some `httr2` to be able to use my redirect url.

For example `oauth_flow_auth_code_listen` function has line:
```
server <- httpuv::startServer(host_ip, port, list(call = listen))
```
but I need to start the erver on the url http://localhost:4321/redirect

I am not sure how to add `/redirct` part to the url.

Than, in part:
```
if (!identical(env$PATH_INFO, "/")) {
```
I have to add redirect too:
```
if (!identical(env$PATH_INFO, "/redirect")) {
```
